### PR TITLE
docs(workflow): add guide on workflow implementation (WIP)

### PR DESCRIPTION
**Pull request by czosel**

### CODE REVIEW

**Comment by winged** [comment]
Just a note (Didn't know myself until yesterday): chore(XX) will get ignored in the changelog, but if you change the type/scope to docs(...), it WILL show up. Depending on how much docs you add, we may want to have this show up in the changelog/release notes.

**Comment by czosel** [comment]
Thanks, TIL!

**Comment by anehx** [comment]
Whats the status here? I think incomplete documentation is better than no documentation right?

**Comment by czosel** [comment]
This still contains some TODOs - but I agree. Feel free to merge.

**Comment by anehx** [comment]
The right commit type and scope would be nice :wink:

**Comment by czosel** [comment]
done :+1:

File: docs/workflow-implementation.md [hunk 1]
```diff
@@ -0,0 +1,153 @@
+## 2. Implementation
+
+In this chapter, we'll create and execute the example "exam" workflow which has been introduced in the [previous chapter, "Workflow Concepts"](workflow-concepts.md). We'll do so by executing mutations on the Caluma GraphQL API - to get most out of this guide, we recommend to follow along using the interactive GraphiQL console of your local Caluma installation (see the [Getting started guide](https://github.com/projectcaluma/caluma#getting-started) for installation instructions).
+
+### Step 1: Workflow Design
+
+In this first step, we'll create the "exam" workflow including its tasks and flows. Although it would feel most natural to start by creating the workflow itself, we'll start by creating the "fill out exam" task - the reason for this is that when creating a new workflow, we have to tell caluma which task(s) to start when the workflow is started.
+
+#### 1a) Create "fill out exam" task
```
Comment (anehx) [inline docs/workflow-implementation.md]: This is not the same level as "1b" and "1c", also it is somehow not rendered as title on GH

Comment (czosel) [reply docs/workflow-implementation.md]: fixed

Comment (czosel) [reply docs/workflow-implementation.md]: ah wait looks like i broke "1d"

Comment (czosel) [reply docs/workflow-implementation.md]: weird ... maybe because it's right after a code block?

**Pull request merged**